python/lsst/ip/isr/isrTask.py

```diff
@@ -918,22 +913,31 @@ def readIsrData(self, dataRef, rawExposure):
                         if self.config.doFlat else None)
 
         brighterFatterKernel = None
+        brighterFatterGains = None
         if self.config.doBrighterFatter is True:
-
-            # Use the new-style cp_pipe version of the kernel is it exists.
             try:
+                # Use the new-style cp_pipe version of the kernel is it exists
+                # If using a new-style kernel, always use the self-consistent
+                # gains, i.e. the ones inside the kernel object itself
                 brighterFatterKernel = dataRef.get("brighterFatterKernel")
+                brighterFatterGains = brighterFatterKernel.gain
+                self.log.info("New style bright-fatter kernel (brighterFatterKernel) loaded")
             except NoResults:
-                # Fall back to the old-style numpy-ndarray style kernel if necessary.
-                try:
+                try:  # Fall back to the old-style numpy-ndarray style kernel if necessary.
                     brighterFatterKernel = dataRef.get("bfKernel")
+                    self.log.info("Old style bright-fatter kernel (np.array) loaded")
                 except NoResults:
                     brighterFatterKernel = None
             if brighterFatterKernel is not None and not isinstance(brighterFatterKernel, numpy.ndarray):
-                # If the kernel is not an ndarray, it's the cp_pipe version, so extract the kernel for
-                # this detector, or raise an error.
+                # If the kernel is not an ndarray, it's the cp_pipe version
+                # so extract the kernel for this detector, or raise an error
                 if self.config.brighterFatterLevel == 'DETECTOR':
-                    brighterFatterKernel = brighterFatterKernel.kernel[ccd.getId()]
+                    if brighterFatterKernel.detectorKernel:
```

Comment:
Have the format of the new-style kernels changed?  This seems to be using different attributes.

Reply:
Yup, exactly - that's the bitrot referred to in the ticket title (everything else is just scope-creep while I was in there in fact 😳)